lib/wit/gitrepo.py

```diff
@@ -48,6 +49,34 @@ class GitRepo:
     def __init__(self, name, wsroot: Path):
         self.name = name
         self.path = wsroot / name
+        # Cache known hashes for quick lookup
+        self._known_hashes = set()  # type: Set[str]
```

Comment:
Do you know about [`@lru_cache`](https://docs.python.org/3.5/library/functools.html#functools.lru_cache), and if not, do you think it'd be better to use that instead of building your own caching utility?

Reply:
I did not know about `lru_cache`, but I did totally think about writing my own decorator to get rid of more boilerplate hahaha, I'll check it out.